augur/io/sequences.py

```diff
@@ -121,3 +122,22 @@ def write_records_to_fasta(records, fasta, seq_id_field='strain', seq_field='seq
                 for key, value in record.items()
                 if key != seq_field
             }
+
+
+class Sequences(StrainBasedFile):
+    """Represents a sequences file."""
+
+    def __init__(self, file:str, id_name:str=None, format:str=None):
+        super().__init__(file, id_name)
+
+        supported_formats = {"fasta"}
+        default_format = "fasta"
+
+        self.format = format or default_format
+        """Sequences file format (e.g. FASTA)."""
+
+        if self.format not in supported_formats:
+            raise AugurError(f"Format {format} not supported.")
+    
+    def get_seqrecords(self) -> Iterable[Bio.SeqIO.SeqRecord]:
+        return Bio.SeqIO.parse(self.file, self.format)
```

Comment:
Instead of an awkwardly-named `get_seqrecords()` method, what about making instances directly iterable? Something like:

```suggestion
    def __iter__(self) -> Iterable[Bio.SeqIO.SeqRecord]:
        return Bio.SeqIO.parse(self.file, self.format)
```

(Also, file needs a newline at the end.)

Reply:
Good idea, done in 3e9b95d859cb9abcb50c75e2792cde75ca47498e...cfddb8ce6a42e55930177fc48abe88b8484ceb64.